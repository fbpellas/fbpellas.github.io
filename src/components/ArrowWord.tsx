import * as React from 'react';

export interface ArrowWordProps {
  word?: string;
  isUp?: boolean;
}

const ArrowWord: React.FC<ArrowWordProps> = (props) => {
  const { word = '', isUp = false } = props;

  const arrow = isUp ? '➚' : '➘';

  return (
    <>
      <div className="inline-period yellow">{arrow}</div>
      {word && word}
    </>
  );
}

export { ArrowWord }