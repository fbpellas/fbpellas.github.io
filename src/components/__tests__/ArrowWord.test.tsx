import React from 'react';
import { render } from '@testing-library/react';
import { ArrowWord } from '../ArrowWord';

test('ArrowWord', () => {
  const { container, rerender } = render(<ArrowWord />);
  expect(container).toMatchSnapshot('down');

  rerender(<ArrowWord isUp />)
  expect(container).toMatchSnapshot('up');

  rerender(<ArrowWord word='something' />)
  expect(container).toMatchSnapshot('word and down');

  rerender(<ArrowWord isUp word='something' />)
  expect(container).toMatchSnapshot('word and up');
});