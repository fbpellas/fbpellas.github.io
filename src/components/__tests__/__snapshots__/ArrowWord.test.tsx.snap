// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowWord: down 1`] = `
<div>
  <div
    class="inline-period yellow"
  >
    ➘
  </div>
  
</div>
`;

exports[`ArrowWord: up 1`] = `
<div>
  <div
    class="inline-period yellow"
  >
    ➚
  </div>
  
</div>
`;

exports[`ArrowWord: word and down 1`] = `
<div>
  <div
    class="inline-period yellow"
  >
    ➘
  </div>
  something
</div>
`;

exports[`ArrowWord: word and up 1`] = `
<div>
  <div
    class="inline-period yellow"
  >
    ➚
  </div>
  something
</div>
`;
